lib/APMSpanContext.js

```diff
@@ -15,11 +15,20 @@
  * limitations under the License.
  */
 
+import { Trace } from './Trace';
+
 class APMSpanContext {
-    constructor(spanId, traceId, parentId) {
+    constructor(spanId, traceId, parentId, parentSpanContext) {
         this._spanId = spanId;
         this._traceId = traceId;
         this._parentId = parentId;
+
+        /**
+         * Hawkular APM related
+         */
+        this._trace = parentSpanContext ? parentSpanContext.getTrace() : new Trace();
+        this._businessTransaction = parentSpanContext ? parentSpanContext.getBusinessTransaction() : undefined;
```

Comment:
Minor thing - could you change these to `transaction` and `getTransaction()` rather than 'business' prefix, as I will probably change this in the model before next release, so that the general term used is 'transaction' - reason is that there are two terms used for the same concept (user defined transactions and business transactions).